### PR TITLE
add endpoint for patching/putting the info of a column

### PR DESCRIPTION
**Pull request by loucadufault**

### CODE REVIEW

File: kanbananza-backend/src/main/controllers/column.js [hunk 1]
```diff
@@ -67,10 +67,51 @@ const index = async (req, res, next) => {
...
+        })
+      );
+    }
+
+    const updatedInfo = {
+      name: req.body.name !== undefined ? req.body.name : column.name,
+      boardId:
+        req.body.boardId !== undefined ? req.body.boardId : column.boardId,
+      order: req.body.order !== undefined ? req.body.order : column.order,
+    };
```
Comment (loucadufault) [inline kanbananza-backend/src/main/controllers/column.js:96]: harmonize between put where all info is provided and patch where provided info is partial

File: kanbananza-backend/src/main/controllers/column.js [hunk 2]
```diff
@@ -67,10 +67,51 @@ const index = async (req, res, next) => {
...
+
+    console.log("service call");
```
Comment (loucadufault) [inline kanbananza-backend/src/main/controllers/column.js:98]: remove

File: kanbananza-backend/src/main/routes/column.js [hunk 3]
```diff
@@ -11,4 +11,6 @@ export default (router) => {
   router.get("/columns", columnController.select);
   // router.get("/columns/:id/cards", columnController.selectCards);
   router.get("/column/:id", idParamValidator, columnController.index);
+  router.put("/column/:id", idParamValidator, validateSchema("createColumn"), columnController.update); // see https://stackoverflow.com/a/34400076
+  router.patch("/column/:id", idParamValidator, columnController.update);
```
Comment (loucadufault) [inline kanbananza-backend/src/main/routes/column.js:15]: standardize the schema names, this should be renamed to "[resource]Info" as it is used for update (sometimes, put but not patch) and create

File: kanbananza-backend/src/main/services/column.js [hunk 4]
```diff
@@ -10,6 +12,14 @@ const createColumn = async ({ name, boardId, order }) => {
     });
   }
 
+  if (!isValidObjectId(boardId)) {
```
Comment (loucadufault) [inline kanbananza-backend/src/main/services/column.js:15]: change to use util method

File: kanbananza-backend/src/main/utils/validators.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import mongoose from "mongoose";
+
+export const isValidMongooseObjectId = (idString) => {
+  if (
+    mongoose.isValidObjectId(idString) &&
```
Comment (loucadufault) [inline kanbananza-backend/src/main/utils/validators.js:5]: unfortunately this is not sufficient

File: kanbananza-backend/src/main/controllers/schemas/user.json [hunk 6]
```diff
@@ -1,30 +1,27 @@
...
-        "isNotEmpty": true,
-        "description": "password of the user"
-      }
+  "description": "all modelled properties of a user",
+  "type": "object",
+  "properties": {
+    "firstName": {
+      "type": "string",
+      "isNotEmpty": true,
+      "description": "first name of the user"
```
Comment (RemiCarriere) [inline kanbananza-backend/src/main/controllers/schemas/user.json:8]: These changes need be made to card.json as well to fix the create card end point.

**Review by RemiCarriere — CHANGES_REQUESTED** [review]
LGTM! Just update the card schema and rebase to fix the failing acceptance tests!

File: kanbananza-backend/src/main/controllers/schemas/user.json (hunk 6 above)
Comment (loucadufault) [reply kanbananza-backend/src/main/controllers/schemas/user.json:8]: yep thanks!

**Pull request merged**